### PR TITLE
Replace `List<T>` with `MyList<T>`

### LINKED ISSUE #7884: List<string> does not contain a definition for Changed
List&lt;string&gt; names = new List&lt;string&gt;();
names.Changed += new EventHandler(ListChanged);
List&lt;T&gt; does not contain a definition for Changed. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8181f41d-b6ce-ad66-084e-8d2b514b87e2
* Version Independent ID: 494c4344-b752-4f86-5b66-1ef4b81c1e75
* Content: [Classes and Objects in C# - A tour of the C# Language](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/classes-and-objects#other-function-members)
* Content Source: [docs/csharp/tour-of-csharp/classes-and-objects.md](https://github.com/dotnet/docs/blob/master/docs/csharp/tour-of-csharp/classes-and-objects.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #7884

The `List<T>` example causes confusion with the .NET Standard type `System.Collections.Generic.List<T>`

Change the name of that type, update all references to the type name, and add a note to the .NET Standard type.

Depends on dotnet/samples#785

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and re-opening for a new build.

File: docs/csharp/tour-of-csharp/classes-and-objects.md [hunk 1]
```diff
@@ -238,9 +241,9 @@ A get accessor corresponds to a parameterless method with a return value of the
 
 A set accessor corresponds to a method with a single parameter named value and no return type. When a property is referenced as the target of an assignment or as the operand of ++ or --, the set accessor is invoked with an argument that provides the new value.
 
-The `List<T>` class declares two properties, Count and Capacity, which are read-only and read-write, respectively. The following is an example of use of these properties.
+The `MyList<T>` class declares two properties, Count and Capacity, which are read-only and read-write, respectively. The following is an example of use of these properties.
```
Comment (mairaw) [inline docs/csharp/tour-of-csharp/classes-and-objects.md]:
```suggestion
The `MyList<T>` class declares two properties, `Count` and `Capacity`, which are read-only and read-write, respectively. The following is an example of use of these properties:
```

File: docs/csharp/tour-of-csharp/classes-and-objects.md [hunk 2]
```diff
@@ -186,45 +186,48 @@ An *abstract method* is a virtual method with no implementation. An abstract met
...
 ## Other function members
 
 Members that contain executable code are collectively known as the *function members* of a class. The preceding section describes methods, which are the primary kind of function members. This section describes the other kinds of function members supported by C#: constructors, properties, indexers, events, operators, and finalizers.
 
-The following shows a generic class called List\<T>, which implements a growable list of objects. The class contains several examples of the most common kinds of function members.
+The following shows a generic class called `MyList<T>`, which implements a growable list of objects. The class contains several examples of the most common kinds of function members.
 
-[!code-csharp[ListClass](../../../samples/snippets/csharp/tour/classes-and-objects/ListBasedExamples.cs#L4-L89)]
+> [!NOTE]
+> This example creates a `MyList` class, which is not the same as the .NET standard <xref:System.Collecions.Generic.List%0601?displayProperty=nameWithType>. It does illustrate the concepts needed for this tour, but is not a replacement for that class.
```
Comment (mairaw) [inline docs/csharp/tour-of-csharp/classes-and-objects.md]: given that you changed the name, do you need this note? 

if you're keeping this, then fix 
Collecions -> Collections

**Review by mairaw — APPROVED** [review]
LGTM. Left some comments for you to consider

File: docs/csharp/tour-of-csharp/classes-and-objects.md (hunk 2 above)
Comment (BillWagner) [reply docs/csharp/tour-of-csharp/classes-and-objects.md]: I'd like to keep it, both to avoid confusion and point people to the reference.

**Pull request merged**